yt/data_objects/static_output.py

```diff
@@ -823,6 +824,8 @@ def _setup_coordinate_handler(self, axis_order: Optional[AxisOrder]) -> None:
             self.no_cgs_equiv_length = True
         elif self.geometry is Geometry.SPECTRAL_CUBE:
             cls = SpectralCubeCoordinateHandler
+        elif self.geometry == "cartesian_subject":
```

Comment:
OK!  This is an actual error.  We need to do two things:

1. Change this to `self.geometry is Geometry.CARTESIAN_SUBJECT`
2. Add `CARTESIAN_SUBJECT = auto()` in the class `Geometry` in `yt/geometry/geometry_enum.py`.

Reply:
3) Make sure this new geometry is acknowledged in all blocks that exhaust the enum (they'll be automatically tagged by mypy)